### PR TITLE
GitHub Action and Encoding logic.

### PR DESCRIPTION
**Pull request by yliuuuu**
*Issue #, if available:* N/A

*Description of changes:*
1. Ported the original implementation to another branch called `docker`. 
    - Now the main branch only holds Rust code. 
2. Added a GitHub Action to auto PR the generated WASM to `react-website` branch. 
3. Added the Session Generation logic in rust. 
4. Change the Readme. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/encode.rs
Comment (am357) [inline src/encode.rs:1]: Looks like there are some clippy warnings such as the following:

```bash
 unused imports: `Error`, `format`

warning: unused imports: `Error`, `format`
 --> src/encode.rs:1:16
  |
1 | use std::fmt::{Error, format};
  |                ^^^^^  ^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
```

File: .github/workflows/pr_wasm_to_wesite.yml
Comment (am357) [inline .github/workflows/pr_wasm_to_wesite.yml:1]: `pr_wasm_to_website.yml`?

**Review by am357 — COMMENTED** [review]
Thanks for the clean-up and also all the good changes!
Can the server (and docker) functionality gets moved to a sub-directory and remain in `main`; this seems to be useful feature to keep around.

File: src/encode.rs
Comment (yliuuuu) [reply src/encode.rs:1]: have run `cargo fix`. should have resolved all compiler warning.

**Review by am357 — APPROVED** [review]

**Pull request merged**